File: backend/booking-service/src/main/java/com/tripify/booking_service/dto/PassengerResponseDTO.java
package com.tripify.booking_service.dto;

// Vista pubblica di un Passenger, compreso il suo "biglietto": qrCodeData resta
// null finché CheckInService non apre il check-in per quella riga (24h prima
// per gli hotel, subito dopo la conferma per voli/attività senza una data di
// check-in propria) - l'app mostra "check-in non ancora aperto" finché è null.
public record PassengerResponseDTO(
        Long id,
        String firstName,
        String lastName,
        String documentType,
        String documentNumber,
        String qrCodeData,
        boolean checkedIn
) {}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/PassengerResponseDTO.kt
package com.tripify.tripify_android.data.model

data class PassengerResponseDTO(
    val id: Long,
    val firstName: String,
    val lastName: String,
    val phoneNumber: String?,
    val taxCode: String,
    val documentType: String,
    val documentNumber: String,
    val qrCodeData: String?,
    val checkedIn: Boolean
)
